src/robotFactory.js

```diff
@@ -1,10 +1,82 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    this.name = name;
+    this.weight = weight;
+    this.chipVersion = chipVersion;
+    this.coords = coords;
+    this.coords.x = coords.x || 0;
+    this.coords.y = coords.y || 0;
+  }
 
-class FlyingRobot {}
+  goForward(step = 1) {
+    if (step > 0) {
+      this.coords.y += step;
+    }
+  }
 
-class DeliveryDrone {}
+  goBack(step = 1) {
+    if (step > 0) {
+      this.coords.y -= step;
+    }
```

Comment:
you do not need this check in all move methods in this task.
```suggestion
      this.coords.y -= step;
```